flaskbb/forum/views.py

```diff
@@ -518,6 +569,10 @@ class EditPost(MethodView):
 
     def get(self, post_id):
         post = Post.query.filter_by(id=post_id).first_or_404()
+
+        if post.topic.first_post_id == post.id:
```

Comment:
Hm, seems like either post or topic should have a `is_first_post` method, maybe both and one calls the other:

```python
class Topic:
    def is_first_post(self, post):
        return self.first_post_id == post.id

class Post:
    def is_first_post(self):
        return self.topic.is_first_post(self)
```

Something like that maybe.